test/unit/string.spec.ts

```diff
@@ -233,6 +233,17 @@ test.each(["hello test"])("string.toUpperCase (%p)", inp => {
     expect(result).toBe(inp.toUpperCase());
 });
 
+test.each(["hello", "test", "hello test"])("string.toUpperCase (union) (%p)", inp => {
```

Comment:
1. This test isn't really related to string.toUpperCase, but to general array string detection, so it should be be placed before `string constrained generic foreach (%p)`  and be renamed
2. There's no need for test.each, since it's not the main reason for this test
3. The code inside template is indented with 2 spaces instead of 4
4. Variables shouldn't begin with underscore, typescript uses it for it's unused diagnostics
5. I don't see much reason to over-complicate test code with inference logic, something like this would be enough to prove test's point:

```ts
const union: "foo" | "bar" = "foo";
return union.length;
```

Reply:
1. I was mainly concerned with `string.*` being transpiled correctly, hence I thought it made sense to have it around here. I'll move it :)
2. I guess you are right.
3. Woopsie 👍 
4. Gotcha - fixed in the next push.
5. Your proposal doesn't cover this. `union` here is just a string literal which already supports `.length`. Enforcing union to be a string type triggers the type guard which appears to compare the string to a union of each or-condition. Inside the if-block, `union` is now inferred as an actual union of strings.
    ```
    const union: string = "foo";

    if (union === "foo" || union === "bar") {
        return union.length;
    }

    return 0;
    ```
    ^ does the trick, though!

Reply:
Oh, looks like TS ignores type annotation in this case and only something like `const union = "foo" as "foo" | "bar";` works